src/analysis/inner_product_space/lax_milgram.lean

```diff
@@ -0,0 +1,182 @@
+/-
+Copyright (c) 2022 Daniel Roca González. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Roca González
+-/
+import analysis.inner_product_space.projection
+import analysis.inner_product_space.dual
+import analysis.normed_space.banach
+
+/-!
+# The The Lax-Milgram Theorem
+
+
+We consider an Hilbert space `V` over `ℝ`
+equipped with a bilinear form `B : V →L[ℝ] V →L[ℝ] ℝ`.
+For every such form we define the
+`lax_milgram_map : V →L[ℝ] V`,
+which is defined by the property `B (lax_milgram_map B v) w = ⟪v, w⟫`
+using the Fréchet-Riesz representation theorem.
+
+Under the hypothesis that `B` is *coercive* (fulfills `is_coercive`),
+we prove the Lax-Milgram theorem:
+that is, the `lax_milgram_map` can be upgraded to a continuous equivalence `lax_milgram`
+
+## References
+
+TODO cite the slides
+
+
+## Tags
+
+dual, Lax-Milgram
+-/
+
+noncomputable theory
+
+
+
+open_locale classical complex_conjugate
+open is_R_or_C continuous_linear_map linear_map linear_equiv
+
+universe u
+def is_coercive
+  {V : Type u} [inner_product_space ℝ V] [complete_space V] (B : V →L[ℝ] V →L[ℝ] ℝ) : Prop :=
+∃ C, (0 < C) ∧ ∀ u, C * ∥u∥ * ∥u∥ ≤ B u u
+
+namespace lax_milgram
+variables {V : Type u} [inner_product_space ℝ V] [complete_space V]
+variables (B : V →L[ℝ] V →L[ℝ] ℝ)
+
+
+def lax_milgram_map (B : V →L[ℝ] V →L[ℝ] ℝ) : V →L[ℝ] V :=
+(inner_product_space.to_dual ℝ V).symm.to_continuous_linear_equiv.to_continuous_linear_map ∘L B
+
+
+@[simp]
+lemma lax_milgram_map_apply (v w : V) : B v w = inner (lax_milgram_map B v) w :=
+by {dunfold lax_milgram_map, simp,}
+
+lemma unique_lax_milgram_map (v f : V)
+  (is_lax_milgram : (∀ w, inner f w = B v w)) :
+  f = lax_milgram_map B v :=
+begin
+  refine inner_product_space.ext_inner_right ℝ _,
+  intro w,
+  rw ←lax_milgram_map_apply,
+  exact is_lax_milgram w,
+end
+
+variables {B}
+
+lemma bounded_below (coercive : is_coercive B) :
+  ∃ C, 0 < C ∧ ∀ v, C * ∥v∥ ≤ ∥lax_milgram_map B v∥ :=
+begin
+  rcases coercive with ⟨C, C_ge_0, coercivity⟩,
+  use C,
+  split, exact C_ge_0,
+  intro v,
+  by_cases h : 0 < ∥v∥,
+  begin
+    have squared : C * ∥v∥ * ∥v∥ ≤ ∥lax_milgram_map B v ∥ * ∥v∥ :=
```

Comment:
Move your `refine` up here and you can be one line shorter:
```suggestion
    refine (mul_le_mul_right h).mp _,
```
(or probably 
```suggestion
    rw mul_le_mul_right h
````
also works)